File: JavaScript/uri-1043.js
// URI - 1043: https://www.urionlinejudge.com.br/judge/pt/problems/view/1043

const input = require('fs').readFileSync('input.txt', 'utf-8');
const values = input.split(' ').map(value => parseFloat(value));

if (checkIfIsTriangle(values))
  console.log('Perimetro = ' + calculateTrianglePerimeter(values).toFixed(1));
else
  console.log('Area = ' + calculateTrapezoidArea(values).toFixed(1));

function checkIfIsTriangle(sides) {
  if (isTriangle(sides))
    return true;
  else
    return false;
}

function isTriangle(sides) {
  return sides[0] + sides[1] > sides[2] 
      && sides[0] + sides[2] > sides[1] 
      && sides[1] + sides[2] > sides[0];
}

function calculateTrapezoidArea(sides) {
  return ((sides[0] + sides[1]) / 2) * sides[2];
}

function calculateTrianglePerimeter(sides) {
  return sides[0] + sides[1] + sides[2];
}

File: Ruby/uri-1043.rb
# frozen_string_literal: true

# URI - 1043: https://www.urionlinejudge.com.br/judge/pt/problems/view/1043

def calculate_area_or_perimeter(values)
  if triangle?(values)
    "Perimetro = #{values[0] + values[1] + values[2]}"
  else
    "Area = #{(((values[0] + values[1]) * values[2]) / 2)}"
  end
end

def triangle?(values)
  (values[0] < values[1] + values[2]) &&
    (values[1] < values[0] + values[2]) &&
    (values[2] < values[0] + values[1])
end

values = gets.split(' ').map(&:to_f)
puts calculate_area_or_perimeter(values)
